### ISSUE TITLE
Spike: NCSC - Additional security for Admin and Developer Devices

### ISSUE THREAD

**Issue by SimonPPledger** [ministryofjustice/modernisation-platform#6723]
### User Story

Following on from the NCSC reviews there are a number of recommendations that we need to further explore.

**Admin and Developer Devices**
AWS accounts hosting the Modernisation Platform are accessed using standard issue MoJ devices. The security posture of these devices was not discussed, however, if these devices render untrusted content through on-device internet browsing or email access, they could be at a higher risk of compromise. Due to this, using a standard issue MoJ device for Administrators and Developers could present an undue risk to the Modernisation Platform and the services it hosts.

**NCSC Recommendation**: Implement technical controls to ensure that AWS accounts are only accessed from devices with a known good security posture. 

**MoJ Response:** The normal access route is via MoJ MACs, which are managed, including logging of actions - eg attempted removal of security features. But it is possible to access via personal machines. Action: We could limit access to MoJ devices only (potentially allowing read-only access from other devices. How would this work for collaborators who might not have an MoJ device?)

This ticket is to look at our suggested action (We could limit access to MoJ devices only (potentially allowing read-only access from other devices)), and to understand whether this would address their risk, whether it would have disadvantages and to get agreement as to what to do

### Value / Purpose

Potentially reduces the platform risk

### Definition of Done

-  [ ] Possible option to reduce this risk, including any negative impact to the service we provide
- [ ] Agreement on whether to action
- [ ] If required, ticket raised to implement

**Comment by mikereiddigital** [comment]
I will be looking at how we can use globalprotect as a sourceIP to limit IAM actions given that all moj devices (DOM1 devices, macbooks) all use that filter.

**Comment by mikereiddigital** [comment]
# Limit access to Modernisation Platform only from MoJ Devices.

## Background

Issue 6273 sets out a requirement to look at how access to the Modernisation Platform can be limited from only MoJ supplied end-user devices as used by MoJ Digital and other Service Support Staff. These would comprise EUCS-provided DOM1 Windows devices that connect from the MoJ networks and MacBooks that would connect from the internet.

These connections would include:

> Access to platform accounts via the AWS console.
> SSM & SSH connections to EC2 instances
> Shell commands using CLI or AWS api calls.
> AWS SDK when using APIs such as BOTO3 or Package SDK that are run from the device.

Not included are:

> OIDC connections from Guthub workflows. It is assumed that github runners operate in a secure environment.
> CLI or other calls to AWS services from EC2s or similar resources hosted in the Modernisation Platform.
> Cross-account access from resources other AWS accounts outside of the Modernisation Platform.

## Proposal

The proposed solution is to limit access to the platform based on the connection’s SourceIP using deny conditions in the relevant IAM policies against the list of known outgoing addresses from the various MoJ networks and those of the GlobalProtect infrastructure. 

This has the benefit of not requiring any additional software or configurations on the user device which would be a significant hurdle given the known effort and time required to implement such a change.

The example below shows how this can be implemented using a documented approach (https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_deny-ip.html)

```
{
    "Version": "2012-10-17",
    "Statement": {
        "Effect": "Deny",
        "Action": "*",
        "Resource": "*",
        "Condition": {
            "NotIpAddress": {
                "aws:SourceIp": [
                    "192.0.2.0/24",
                    "203.0.113.0/24"
                ]
            }
        }
    }
}
```

Unlike the above example, it is suggested that we would manage the IPs & CIDRs as a json file given their number.

## Required Changes to GlobalProtect Filter Rules

GlobalProtect is not a VPN but rather a Proxy that requires rules that determine whether the outgoing traffic from the user’s device is sent through the GP infrastructure and so out via its gateway & thus obtaining one of its outgoing addresses. To this end we would need to raise a change request with the Technology Services team to have added the rules for the required URLs, including AWS endpoints.

These URLs would include:

> URLs used for Modernisation Platform console access such as moj.awsapps.com
> Endpoints addresses but using wildcards e.g. *.amazonaws.com or limited to regions e.g *.eu-west-2.amazonaws.com).
> API endpoint addresses e.g. *.api.aws

The full scope of these would need to be determined and rigorous testing undertaken to confirm that all endpoints have been captured as rules.

It should be noted that as Technology Services operates an ITIL-like change management process, the above request would need to engage with the TS-teams to ensure its progress and timely implementation.

**Comment by mikereiddigital** [comment]
Outcomes & actions from the review earlier today of the above proposal:

1. The proposal as described was challenged given the risks as described. Specifically:

- That the GlobalProtect infrastructure is robust enough to avoid unplanned outages that could result in the outgoing IPs to change. 
- The change control processes for adding new GlobalProtect rules no not prevent timely changes to rules.

Action - follow up with the Technical Services team & confirm the current status of the above & feed back.

2. It was also suggested that device assurance at the point of SSO login via the planned EntraID product is also investigated as an alternative.  

Action - follow up & determine whether the planned EntraID work will be limited to just MoJ issued devices.

3. It was also noted that either of the two approaches would prevent Collaborator type accounts from accessing Modernisation Platform and whether the scope of these restrictions should focus solely on Administrator accounts.

**Comment by mikereiddigital** [comment]
Catch up arranged with Ian Harvey & Martin Cadren on Monday at 10am.

**Comment by mikereiddigital** [comment]
Spoke with Ian as Martin could not make the meeting & have rescheduled for Wednesday. 

- The GP infrastructure has been stable since the start of the year and the outgoing IPs have not changed. 
- The arrangements for adding new rules are the same as before - a change request & a CAB for approval. There is no option to apply these changes via a PR. 

No hard information on the use of Entra ID for non-MOJ devices though will follow this up again on Wednesday.

**Comment by mikereiddigital** [comment]
Having spoken again with Ian Harvey this morning, the following was discussed:

- There is work in place to move away from hosting the PaloAlto infrastructure on AWS to a new cloud-hosted PauloAlto service - Prisma Access (https://www.paloaltonetworks.co.uk/sase/access). The timescales are around 3 months for implementation but there are dependencies with the move to Entra ID.

- In addition work in place to implement client-based authorisation using device-hosted certificates. The current GlobalProtect solution uses the same certificates regardless of the user.

Given these infrastructure changes and the impact it could have on any approach we implement before these changes are in place it is suggested that:

- For Administrator access to Modernisation Platform, the use of outgoing address to filter Administrator Access to the Modernisation Platform once the migration to Prisma Access is assessed once the move been completed and the new service is both operational and stable. This will be to determine whether stable outgoing IPs exist from the service that can be used to reliably filter incoming connections to Administrator accounts.

- In addition, the above assessment will consider processes in place within Technology Services for request to amend filter rules as to whether working with such processes are practical in the context of the ability of the MP team to administer the platform.

-  Feasibility of limiting non-Administrator developer-level access to the Modernisation Platform AWS accounts is re-assessed once the Entra ID solution is in place and we can better understand the use of client-side certificates & whether these can provide the required device assurance.

**Comment by mikereiddigital** [comment]
Note - Unrestricted accounts (e.g. Bichard) are out of scope for this ticket.

**Comment by mikereiddigital** [comment]
Spoke with Simon & Ewa. Agreed that:

- We would implement access controls just for MP team access - see ticket https://github.com/ministryofjustice/modernisation-platform/issues/7536

- Simon will follow up with Matt White for information on EntraID and whether that will provide the means to limit access only from MoJ devices.
